protected/humhub/modules/user/models/User.php

```diff
@@ -323,13 +324,29 @@ public function getGroupUsers()
         return $this->hasMany(GroupUser::class, ['user_id' => 'id']);
     }
 
+    /**
+     * Returns default Group as ActiveQuery
+     * @return \yii\db\ActiveQuery
+     */
+    public function getDefaultGroup()
+    {
+        return Group::find()->where(['is_default_group' => 1]);
+    }
+
     /**
      * Returns all Group relations of this user as ActiveQuery
      * @return \yii\db\ActiveQuery
      */
     public function getGroups()
     {
-        return $this->hasMany(Group::class, ['id' => 'group_id'])->via('groupUsers');
+        $groups = $this->hasMany(Group::class, ['id' => 'group_id'])->via('groupUsers');
+
+        if ($groups->count() == 0) {
```

Comment:
Can we remove this here and show an error if no group is selected in UserEdit?

Reply:
If we will remove this then users without a group will not be fallback to default group. It may happens when we remove some group and this group is used as single group for some users, I even implemented a confirmation for such case today in the commit https://github.com/humhub/humhub/pull/4742/commits/375342f110dc528bc5a194aec3390fd7f70995b4.
I.e. currently we assign a default group to users only on migration and then possible to have users without relation in the table `group_user`.

Reply:
@yurabakhtin On group delete, can we then create actual group memberships to the default group for those users (as in migration)?

Reply:
@luke- yes, we can, I have implemented this in commit https://github.com/humhub/humhub/pull/4742/commits/d8a96dc16ac22b87ddff7697395f99d4b2772d64:

- removed the code `if ($groups->count() == 0) { ...`,
- show an error if no group is selected in UserEdit,
- create memberships to default group on deleting a group for users without any group as in migration.